src/main/java/address/storage/StorageManager.java

```diff
@@ -34,17 +36,17 @@ private void handleLoadDataRequestEvent(LoadDataRequestEvent ofe) {
     }
 
     @Subscribe
-    private void handleLocalModelChangedEvent(LocalModelChangedEvent lmce) {
-        final File targetFile = PreferencesManager.getInstance().getPersonFile();
+    private void handleLocalModelChangedEvent(LocalModelChangedEvent e) {
         System.out.println("Local data changed, saving to primary data file");
-        saveDataToFile(targetFile, lmce.personData, lmce.groupData);
+        EventManager.getInstance().post(new SaveRequestEvent(
+                PrefsManager.getInstance().getSaveLocation(), e.personData, e.groupData));
     }
 
     @Subscribe
-    private void handleLocalModelSyncedEvent(LocalModelSyncedFromCloudEvent lmse) {
-        final File targetFile = PreferencesManager.getInstance().getPersonFile();
+    private void handleLocalModelSyncedEvent(LocalModelSyncedFromCloudEvent e) {
         System.out.println("Local data synced, saving to primary data file");
-        saveDataToFile(targetFile, lmse.personData, lmse.groupData);
+        EventManager.getInstance().post(new SaveRequestEvent(
+                PrefsManager.getInstance().getSaveLocation(), e.personData, e.groupData));
     }
```

Comment:
These two methods are counterparts. So method signatures should be similar, name format wise. e.g. 
`handleLocalModelChangedEvent`
`handleRemoteModelChangedEvent`